HerPublicWebsite.BusinessLogic/ExternalServices/EmailSending/GovUkNotifyApi.cs

```diff
@@ -108,36 +112,31 @@ string followUpLink
         }
         
         public void SendComplianceEmail(
-                MemoryStream recentReferralRequestOverviewFileData,
-                MemoryStream recentReferralRequestFollowUpFileData,
-                MemoryStream historicReferralRequestFollowUpFileData
-                )
-                {
-                    var recipientList = govUkNotifyConfig.ComplianceEmailRecipients;
-                    if (String.IsNullOrEmpty(recipientList))
-                    {
-                        return;
-                    }
-                    
-                    var template = govUkNotifyConfig.ComplianceReportTemplate;
-                    Dictionary<String, dynamic> personalisation = new Dictionary<String, dynamic>
-                    {
-                        { "File1Link", NotificationClient.PrepareUpload(recentReferralRequestOverviewFileData.ToArray(), true)},
-                        { "File2Link", NotificationClient.PrepareUpload(recentReferralRequestFollowUpFileData.ToArray(), true)},
-                        { "File3Link", NotificationClient.PrepareUpload(historicReferralRequestFollowUpFileData.ToArray(), true)},
-                    };
-                    foreach (var emailAddress in recipientList.Split(","))
-                    {
-                        var emailModel = new GovUkNotifyEmailModel
-                        {
-                            EmailAddress = emailAddress.Trim(),
-                            TemplateId = template.Id,
-                            Personalisation = personalisation
-                        };
-                        SendEmail(emailModel);
-                    }
-                    
-                }    
+            MemoryStream recentReferralRequestOverviewFileData,
+            MemoryStream recentReferralRequestFollowUpFileData,
+            MemoryStream historicReferralRequestFollowUpFileData)
+        {
+            var recipientList = govUkNotifyConfig.ComplianceEmailRecipients;
+            var template = govUkNotifyConfig.ComplianceReportTemplate;
+            var personalisation = new Dictionary<string, dynamic>
+            {
+                { template.File1Link, NotificationClient.PrepareUpload(recentReferralRequestOverviewFileData.ToArray(), true)},
+                { template.File2Link, NotificationClient.PrepareUpload(recentReferralRequestFollowUpFileData.ToArray(), true)},
+                { template.File3Link, NotificationClient.PrepareUpload(historicReferralRequestFollowUpFileData.ToArray(), true)},
```

Comment:
I'm concerned about the change from `"File1Link"` to `template.File1Link`, etc.

I believe the latter resolves to `"file 1 link"`, based on `appsettings.json`. So changing the code here actively changes the personalisation keys.

Having checked it in GOV.UK Notify, it looks like `"file 1 link"` should be correct, but I'm concerned about making this change without fuller testing.

For safety, we should leave the keys as they were (`"File1Link"`, etc.).

I will consider whether we should make a change to correct this to all of the proper `…Placeholder` style, but let's not do it in this review.

Reply:
the personalisation key system does seem a bit confusing to me, the process by which the key "File1Link" gets understood by GOV.UK notify as the key "file 1 link" doesn't seem overly clear to me (which is what the previous system did), unless it's just taking an educated guess

for now, yeah it can be left as "File1Link"